crates/uv-settings/src/settings.rs

```diff
@@ -1243,3 +1245,90 @@ impl From<ResolverInstallerOptions> for InstallerOptions {
         }
     }
 }
+
+/// The options persisted alongside an installed tool.
+///
+/// A mirror of [`ResolverInstallerOptions`], without upgrades and reinstalls, which shouldn't be
+/// persisted in a tool receipt.
```

Comment:
We go through some effort to avoid adding `--upgrade` to tool receipts. Does that seem right? Otherwise, `uv tool install black --upgrade` saves `upgrade` in the receipt...

Reply:
Yes that would be weird.